app/src/test/java/com/kickstarter/libs/utils/CodeVerifierTest.kt

```diff
@@ -0,0 +1,129 @@
+package com.kickstarter.libs.utils
+
+import com.kickstarter.KSRobolectricTestCase
+import com.kickstarter.libs.utils.CodeVerifier.Companion.ERROR_DO_NOT_MATCH
+import com.kickstarter.libs.utils.CodeVerifier.Companion.ERROR_TOO_LONG
+import com.kickstarter.libs.utils.CodeVerifier.Companion.ERROR_TOO_SHORT
+import com.kickstarter.libs.utils.CodeVerifier.Companion.MAX_CODE_VERIFIER_ENTROPY
+import com.kickstarter.libs.utils.CodeVerifier.Companion.MIN_CODE_VERIFIER_ENTROPY
+import com.kickstarter.libs.utils.CodeVerifier.Companion.generateCodeChallenge
+import com.kickstarter.libs.utils.CodeVerifier.Companion.generateRandomCodeVerifier
+import org.junit.Assert.assertThrows
+import org.junit.Test
+
+class CodeVerifierTest : KSRobolectricTestCase() {
+
+    @Test
+    fun checkCodeVerifier_tooShort_throwsException() {
```

Comment:
These are great. With the exception of entropy, I can port the exact same checks to iOS as well.